packages/angular_devkit/build_angular/src/utils/index-file/index-html-generator.ts

```diff
@@ -0,0 +1,119 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { join } from 'path';
+import { readFile } from '../fs';
+import { NormalizeOptimizationOptions } from '../normalize-optimization';
+import { stripBom } from '../strip-bom';
+import { CrossOriginValue, FileInfo, augmentIndexHtml } from './augment-index-html';
+import { InlineFontsProcessor } from './inline-fonts';
+
+type IndexHtmlGeneratorPlugin = (html: string, options: IndexHtmlGeneratorProcessOptions) => Promise<string>;
+
+export interface IndexHtmlGeneratorProcessOptions {
+  lang: string | undefined;
+  baseHref: string | undefined;
+  outputPath: string;
+  files: FileInfo[];
+  noModuleFiles: FileInfo[];
+  moduleFiles: FileInfo[];
+}
+
+export interface IndexHtmlGeneratorOptions {
+  indexPath: string;
+  deployUrl?: string;
+  sri?: boolean;
+  entrypoints: string[];
+  postTransform?: IndexHtmlTransform;
+  crossOrigin?: CrossOriginValue;
+  optimization?: NormalizeOptimizationOptions;
+  WOFFSupportNeeded: boolean;
+}
+
+export type IndexHtmlTransform = (content: string) => Promise<string>;
+
+export class IndexHtmlGenerator {
+  private readonly plugins: IndexHtmlGeneratorPlugin[];
+
+  constructor(readonly options: IndexHtmlGeneratorOptions) {
+    this.plugins = [
+      augmentIndexHtmlPlugin(this),
+      this.options.optimization?.fonts.inline ? inlineFontsPlugin(this) : noopPlugin(this),
```

Comment:
could potentially drop the `noopPlugin` if `inlineFontsPlugin` was pushed in an `if` block.

Reply:
Agreed